### DISCUSSION TITLE
Playing tone melody with delays

### DISCUSSION THREAD

**Discussion by ShivamJoker** [contrem/arduino-timer#53] [Q&A]
What would be the ideal way of playing some long list of melodies using this libary

```cpp
  toneTimer.in(0, playTone, 1047);

  toneTimer.in(415, playTone, 1109);

  toneTimer.in(659, playTone, 1047);

  toneTimer.in(1043, playTone, 1109);

  toneTimer.in(1287, playTone, 1047);
 .....
```

Can we keep adding these dynamically ? I am passing the frequency

**Comment by philj404** [comment]
In principle your use of toneTimer.in(...) could work... but it has some drawbacks:

- toneTimer will need to reserve enough storage for ALL notes when you load your tune.  Tunes with many notes will take a lot of space
- playTone() MUST finish before the next scheduled note starts (or your tune timing will stutter)
- playTone() does not have enough information to handle notes of different lengths (not necessarily problem for some tunes)

Note you do not actually need to worry about all the tones at once.  Only one is active at any one time.  You may want to consider using something like:
```
 timer.every(shortestNoteLength, continuePlayingMySong);
```
and have continuePlayingMySong() index through an array of tone frequencies and lengths, start the next note and return ```true``` to keep running.

At this point, there are probably many tutorials about how to play a song with Arduino.  Usually the tutorials only focus one one task (for example play a song).  It should be straightforward (it may not be trivial!) to use an arduino-timer to play songs _and do something else at the same time_.

**Reply by ShivamJoker** [reply]
Thanks for the quick reply.

I can have different frequencies and lengths in two different arrays

But I am not sure what should I do in continuePlaying function, I mean all the tones have different lengths so how will I be calling them at particular times ?
